shavee-bin/src/args.rs

```diff
@@ -128,35 +130,25 @@ impl Sargs {
         let arg = app.get_matches_from_safe(args)?;
 
         // if keyfile arg is entered, then its value will be used
-        // if not, then pass "NULL"
-        // // exceptions should not happen, because the entry is already validated by clap
-        let file = match arg.value_of("keyfile") {
-            Some(s) => s.to_string(),
-            None => "NULL".to_string(),
-        };
+        let file = arg.value_of("keyfile")
+            .map(str::to_string);
 
         // if zset arg is entered, then its value will be used
-        // if not, then pass "NULL"
-        // NOTE: validating dataset is done in zfs.rs module
-        let mut dataset = match arg.value_of("zset") {
-            Some(s) => {
-                let mut s = s.to_string();
-                // remove the trailing '/' from dataset if entered
+        // NOTE: validating dataset is done by zfs module
+        let mut dataset = match arg.value_of("zset")
```

Comment:
Compare this implementation to line 179. Both can be done to manipulate contents wrap with `Option` and it depends on the programmer which one is preferred.

Here it is an example using `match`